### PR TITLE
Support for evaluatedRecords from retrieves

### PR DESCRIPTION
**Pull request by mgramigna**
This PR adds the concept of `evaluatedRecords` to the `Context` object and `Results`. `evaluatedRecords` will contain every record that was processed during a retrieve. When a retrieve occurs, the resulting records are added to the `evaluatedRecords` array on the `Context`. When reporting the results, the `evaluatedRecords` for all library contexts are aggregated together into one array and added to the `Results` object.

This functionality will allow eCQM-focused use cases to see which records were processed by the engine, and report those back to the caller

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [X] Tests are included and test edge cases
- [X] Tests have been run locally and pass
- [X] Code coverage has not gone down and all code touched or added is covered.
- [X] Code passes lint and prettier (hint: use `yarn run test:plus` to run tests, lint, and prettier)
- [X] **N/A** All dependent libraries are appropriately updated or have a corresponding PR related to this change
- [X] `cql4browsers.js` built with `yarn run build:browserify` if source changed.

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Review by cmoesel — COMMENTED** [review]
I dig it.  I made a few suggestions.  They aren't super important, so I won't formally request changes, but I think at least the first one is an actual improvement (but second suggestion is maybe a lateral move).

File: src/elm/external.js [hunk 1]
```diff
@@ -31,6 +31,11 @@ class Retrieve extends Expression {
       records = records.filter(r => range.includes(r.getDateOrInterval(this.dateProperty)));
     }
 
+    if (Array.isArray(records)) {
+      records.forEach(rec => ctx.evaluatedRecords.push(rec));
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply src/elm/external.js]: Yeah good call. You sure you don't want the following though? ;)

``` JavaScript
ctx.evaluatedRecords.push(Array.isArray(records) ? ...records : records);
```

[the remark this answers is not in the document]
Comment (cmoesel) [reply src/elm/external.js]: I actually started to comment that you _could_ do it that way, but then I second-guessed myself because I wasn't sure if a spread operator would work inside a ternary like that.  Does it?

I'm not one of those ternary operator haters.  Are you?

**Review by cmoesel — APPROVED** [review]
Looks good to me!
